state/application.go

```diff
@@ -1737,6 +1745,71 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// SetDownloadedIDAndHash updates the applications charm origin with ID and
+// hash values. This should ONLY be done from the async downloader.
+// The hash cannot be updated if the charm origin has no ID, nor was one
+// provided as an argument. The ID cannot be changed.
+func (a *Application) SetDownloadedIDAndHash(ID, hash string) error {
```

Comment:
Nit: `ID` -> `id` as it's a local variable.

Reply:
I found a few different opinions, doing to keep consistent in juju.

[This rule also applies to "ID" when it is short for "identifier" (which is pretty much all cases when it's not the "id" as in "ego", "superego")](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)